mod_ci/controllers.py

```diff
@@ -123,6 +124,58 @@ def kvm_processor_windows(db, repository, delay):
         db, kvm_name, TestPlatform.windows, repository, delay)
 
 
+@mod_ci.route('/ci_badge')
+def ci_badge():
+    dwg = svgwrite.Drawing()
+    commit = Test.query.filter(Test.test_type != Test.pull_request)
+    release = Test.query.filter(Test.test_type != Test.commit)
+    if kvm_test.name is kvm_processor_linux:
+        if commit.TestStatus = TestStatus.completed and Status.SUCCESS:
+            dwg.add(dwg.image(
+                    "https://img.shields.io/badge/Test-passed-brightgreen.svg",
+                    size=(4, 12))
+                    dwg.save()
+                    elif commit.TestStatus=TestStatus.canceled and Status.ERROR:
+                    dwg.add(dwg.image(
+                        "https://img.shields.io/badge/Test-canceled-lightgrey.svg",
+                        size=(4, 12))
+                    dwg.save()
+                    return
+                    elif commit.TestStatus=TestStatus.completed and Status.FAILURE:
```

Comment:
All these if's make no sense. Determining the status of a test result is more complicated than this.

Besides, where is Status coming from?